main/src/library/Array.flix

```diff
@@ -603,7 +603,7 @@ namespace Array {
                         let pos2 = arrayWrites(pos1, a, out);
                         (pos2, i+1)
                     }};
-                foldLeft(overwrite, (0, 0), arrs);
+                discard foldLeft(overwrite, (0, 0), arrs);
```

Comment:
Use foreach?

Reply:
The function needs to keep track of how big the nested expressions are so I think its best as a fold. It could be a zipWithIndex and then a fold but a fold none the less.